pkg/scm/types.go

```diff
@@ -3,18 +3,37 @@ package scm
 import (
 	"context"
 	"net/http"
+	"regexp"
+	"strconv"
 	"strings"
 
 	"github.com/jippi/scm-engine/pkg/state"
 	"github.com/jippi/scm-engine/pkg/types"
 )
 
 type Actor struct {
+	ID       string
 	Username string
 	Email    *string
 	IsBot    bool
 }
 
+func (a Actor) IntID() int {
+	re := regexp.MustCompile(`(\d+)`)
```

Comment:
gitlab ids return as string in the form `gid://gitlab/User/1`, but requires us to parse as int ids for a few of their api requests, e.g. merge request updates

Reply:
You can alternatively do `strings.TrimPrefix(a.ID, "gid://gitlab/User/")`  as well, which might be faster (and simpler)

Reply:
i did it this way assuming github support, if we avoid using an actor all together and cast the eval context i can just rip this out